docs/variables.md

```diff
@@ -243,7 +243,9 @@ For `Template`-level metrics:
 | Variable | Description|
 |----------|------------|
 | `workflow.name` | Workflow name |
+| `workflow.templateName` | Workflow template name. This works only for template names, it may give wrong answers for non-generated names |
```

Comment:
At the risk of being verbose, I wonder if we should call this `workflowTemplateName`? When I see `templateName`, I think of the individual step, not the WorkflowTemplate

Reply:
From [this](https://argoproj.github.io/argo-workflows/workflow-templates/) link: "The terms WorkflowTemplate and template have created an unfortunate naming collision and have created some confusion in the past."

Reply:
I know what you mean. To fit the two part naming of all the rest of the variables how about workflowTemplate.name?

Reply:
For my understanding, does this also represent the CronWorkflow name for a Workflow generated by a CronWorkflow?

Reply:
@juliev0, yes, it should do. Ideally it would always be able to derive the name from the parent, but as the names are generated and this attempts to reverse that, I'm basing it on what others in #6772 have found works.

Reply:
So, in that case using `workflowTemplate` in the name would also be inaccurate. I'm thinking something like `origin`, or otherwise dividing this into multiple variables. What do you think?

Reply:
So `workflow.origin` is your proposal? Or `workflow.originName` to demonstrate it's the name of something perhaps.

Reply:
`workflow.originName` sounds good to me. If you'd like you can wait a few days to see if anyone chimes in first, for risk of your changing it once and then needing to change it again.